coderd/userauth.go

```diff
@@ -535,9 +536,7 @@ func (api *API) userOAuth2Github(rw http.ResponseWriter, r *http.Request) {
 			}
 		}
 		if len(selectedMemberships) == 0 {
-			httpapi.Write(ctx, rw, http.StatusUnauthorized, codersdk.Response{
-				Message: "You aren't a member of the authorized Github organizations!",
-			})
+			httpmw.RedirectToLogin(rw, r, &url.URL{Path: "/login"}, "You aren't a member of the authorized Github organizations!")
```

Comment:
if we're ok with this pattern, I can redirect similarly on the other oauth errors.

Reply:
One issue with this function is that it sets `?redirect=${currentPath}`, which we don't want to do. If you could change the `httpmw.RedirectToLogin` function to accept a nil `*http.Request` and avoid adding the redirect query param then it should solve the problem